core/config/value.go

```diff
@@ -171,8 +171,18 @@ func (cv Value) TryAsInt() (int, bool) {
 	if val, err := convertValue(v, reflect.TypeOf(0)); v != nil && err == nil {
 		return val.(int), true
 	}
-	return 0, false
-
+	switch val := v.(type) {
+	case int32:
```

Comment:
We can't collapse the switch here. It errors at int(val) and the code needs type assertion.